src/Boost.sol

```diff
@@ -69,62 +74,71 @@ contract Boost is IBoost, EIP712("boost", "1"), Ownable {
 
         uint256 balance = 0;
         if (tokenFee > 0) {
+            // The token fee is calculated and subtracted from the deposit amount to get the initial boost balance
             uint256 tokenFeeAmount = _amount / tokenFee;
+            // tokenFeeAmount < _amount, therefore balance will never underflow
             balance = _amount - tokenFeeAmount;
             tokenFeeBalances[address(_token)] += tokenFeeAmount;
         } else {
+            // When there is no token fee, the boost balance is full deposit amount
             balance = _amount;
         }
 
-        uint256 newId = nextBoostId;
-        nextBoostId++;
-        boosts[newId] = BoostConfig({
-            strategyURI: _strategyURI,
-            token: _token,
-            balance: balance,
-            guard: _guard,
-            start: _start,
-            end: _end,
-            owner: _owner
-        });
+        uint256 boostId = nextBoostId;
+        unchecked {
+            // Overflows if 2**256 boosts are minted
+            nextBoostId++;
+        }
 
+        // Minting the boost as an ERC721 and storing the config data
```

Comment:
This function should be called `mint` instead of `createBoost` now (following common function name for NFTs)